odoo/tools/mail.py

```diff
@@ -380,6 +380,14 @@ def html2plaintext(html, body_id=None, encoding='utf-8'):
             link.text = '%s [%s]' % (link.text, i)
             url_index.append(url)
 
+    for img in tree.findall('.//img'):
+        src = img.get('src')
+        if src:
+            i += 1
+            img.tag = 'span'
+            img.text = '%s [%s]' % (src.split('/')[-1].split('.')[0], i)
```

Comment:
I know that for the url we have that too, but i think that the tag and the text are useless for the image at least so it can be removed :smile: Moreover if we really wanted to keep the text, in my opinion it's not robust enough specially with no test, if for any reason the src is not formatted like this it will put something else has the text for the img :smile:

Reply:
> I know that for the url we have that too, but i think that the tag and the text are useless for the image at least so it can be removed.

No, we can't remove that because we need to add some text in place of the img tag so that it remains after all tags are removed. The html2plaintext function should return something instead of an empty string in cases where company_details only contain image tags.
https://github.com/odoo/odoo/blob/4a13bbc1b9d5f449e51c065fd9530a099d4cfec4/odoo/tools/mail.py#L396

Reply:
As @malb-odoo mentioned, splitting `src` using `/` and `.`, and hoping (because that's all it is) to retrieve the image name is really not a good idea :confused: 
Try using drag and drop to insert the image in the `company_details` field for example :wink: 
(Please update your test to include that use case)
It shouldn't be to hard to do that with a regex :nerd_face: